File: Code/matlab/current.m
function [Jx,Jy,Jz] = current(Data,dx,dy,dz,m)
%Current of three direction
%Data: Wavefunction
%dx,dy,dz: resolution
%m: mass

    hbar = 6.62607004e-34/(2*pi);    
    
    [VxData,VyData,VzData] = gradient(Data,dx,dy,dz);
    [VxData_C,VyData_C,VzData_C] = gradient(conj(Data),dx,dy,dz); 

    Jx = (hbar/(2i*m))*(conj(Data) .*VxData - Data.*VxData_C); 
    Jy = (hbar/(2i*m))*(conj(Data).*VyData - Data.*VyData_C);
    Jz = (hbar/(2i*m))*(conj(Data).*VzData - Data.*VzData_C);
    
% Equal to first one
%     Jx = (hbar/(m))*imag(conj(Data).*VxData); 
%     Jy = (hbar/(m))*imag(conj(Data).*VyData);
%     Jz = (hbar/(m))*imag(conj(Data).*VzData);
    
%Dimensionless equaltion, has slightly difference.
%         Jx = omegaZ*unit^-2*imag(conj(Data).*VxData); 
%         Jy = omegaZ*unit^-2*imag(conj(Data).*VyData);
%         Jz = omegaZ*unit^-2*imag(conj(Data).*VzData);
end

File: Code/python/current.py
from numpy import pi
from numpy import gradient
from numpy import conjugate
    
def current(data,dx: float,dy,dz,m):
    """Calculating probability density current

    Args:
        data: Wavefunction (3D, with unit)
        dx,dy,dz: Difference along 3 dimensions
        m: mass of ato
    Returns:
        Jx, Jy, Jz: Probability density current 
            of three dimensions
    """

    hbar = 6.62607004e-34/(2*pi)
    
    [VxData,VyData,VzData] = gradient(data,dx,dy,dz)
    [VxData_C,VyData_C,VzData_C] = gradient(conjugate(data),dx,dy,dz)

    Jx = (hbar/(2j*m))*(conjugate(data)*VxData - data*VxData_C) 
    Jy = (hbar/(2j*m))*(conjugate(data)*VyData - data*VyData_C)
    Jz = (hbar/(2j*m))*(conjugate(data)*VzData - data*VzData_C)
    
# Equal to first one
#     Jx = (hbar/(m))*imag(conj(data).*VxData) 
#     Jy = (hbar/(m))*imag(conj(data).*VyData)
#     Jz = (hbar/(m))*imag(conj(data).*VzData)
    
#Dimensionless equaltion, has slightly difference.
#         Jx = omegaZ*unit^-2*imag(conj(data).*VxData) 
#         Jy = omegaZ*unit^-2*imag(conj(data).*VyData)
#         Jz = omegaZ*unit^-2*imag(conj(data).*VzData)
    return (Jx, Jy, Jz)